packages/plugin-ext/src/main/browser/workspace-main.ts

```diff
@@ -227,6 +233,59 @@ export class WorkspaceMainImpl implements WorkspaceMain, Disposable {
         return uriStrs.map(uriStr => Uri.parse(uriStr));
     }
 
+    async $findTextInFiles(query: theia.TextSearchQuery, options: theia.FindTextInFilesOptions,
```

Comment:
JSON RPC call cannot accept functions like `callback`. I am not sure how it works. Could you elaborate please?

That's how remote API looks in VS Code: https://github.com/microsoft/vscode/blob/45aa0bf525b08d7e23c9b10ca65f1de55949e601/src/vs/workbench/api/common/extHost.protocol.ts#L722

Reply:
Right, this _callback does not have a value it is `undefined` I guess we can remove this parameter. I did not do that, because I saw on the VSCode impl that they do allow to send a function like this. The workaround I added was the usage of this.proxy.$onTextSearchResult instead of using this _callback.

Once we are on $onTextSearchResult it process the result and emit an event `searchInWorkspaceEmitter` that is being listened to run the original `callback`.

```
        const disposable = this.searchInWorkspaceEmitter.event(result => {
            callback(result);
        });
        if (token) {
            token.onCancellationRequested(() => {
                disposable.dispose();
            });
        }
```

Reply:
ok, please remove unused arguments

Reply:
How does it work with 2 parallel searches? Results reported for both calls, should not we filter them? It would be good to have a way to test it.

Reply:
Ok, just remove the _callback argument and you are right we should filter parallel searches, just added a way to do that. I guess that a way to test would just run the test case two at the same time with different queries. The approach is to have a sequence number for the search api request and filter the results based on that.